shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/networking/OpenshiftRouteSpecMatchesHelper.java

```diff
@@ -0,0 +1,86 @@
+package com.redhat.service.smartevents.shard.operator.networking;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+import io.fabric8.kubernetes.api.model.IntOrString;
+import io.fabric8.openshift.api.model.RouteSpec;
+
+public class OpenshiftRouteSpecMatchesHelper {
+
+    private OpenshiftRouteSpecMatchesHelper() {
+        //Static utility class
+    }
+
+    public static boolean matches(RouteSpec existing, RouteSpec expected) {
+        Function<RouteSpec, RouteSpec> spec = (RouteSpec source) -> source;
+        if (!Objects.equals(spec.apply(existing), spec.apply(expected))) {
+            return false;
+        }
```

Comment:
This check returns false even when existing and expected don't match in a parameter which we don't check (i.e. `to.weight`).

Reply:
@sutaakar Fixed.. I made a right mess of this change. Hopefully it's OK now.